File: OOAD/shopping-cart-app/src/shopping/components/Product.java
package shopping.components;

public class Product {
	private static int totalNumberOfId;
	
	private int id;
	private String name;
	private double price;
	private double discountPercent;
	
	
	static{
		totalNumberOfId=100;
	}
	public Product(String name,double price,double discountPercentage){
		this.name=name;
		this.price=price;
		this.discountPercent=discountPercentage;
		id=generateId();
	}
	
	public int generateId(){
		return (++totalNumberOfId);
	}
	
	public double calcDiscountCost(){
		
		return (price-(price*(discountPercent*0.01)));
	}
	
	public String getName(){
		return name;
	}
	
	public int getProductId(){
		return id;
	}
	
	public String toString(){
		return "Product Id ="+getProductId()+"\nName of product = "+getName()+"\nPrice = "+calcDiscountCost();
	}
}

File: C#/MVC/ShopingCart-App/ShopingCart-App/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ShopingCart_App.Models
{
    public class Product
    {
        private static int totalNumberOfId;

        private int id;
        private string name;
        private double price;

        static Product()
        {
            totalNumberOfId = 100;
        }

        public Product()
        {

        }

        public Product(string name, double price)
        {
            this.ProductName = name;
            this.ProductPrice = price;
            ProductId = generateId();
        }

        public int ProductId
        {
            get
            {
                return id;
            }

            set
            {
                id = value;
            }
        }

        public string ProductName
        {
            get
            {
                return name;
            }

            set
            {
                name = value;
            }
        }

        public double ProductPrice
        {
            get
            {
                return price;
            }

            set
            {
                price = value;
            }
        }

        public int generateId()
        {
            return (++totalNumberOfId);
        }

        public override string ToString()
        {
            return "Product Id =" + ProductId + "\nName of product = " + ProductName + "\nPrice = " + ProductPrice;
        }
    }
}